templates/puppet.spec

```diff
@@ -0,0 +1,52 @@
+%{!?upstream_version: %global upstream_version %{version}%{?milestone}}
+{% if metadata.from_puppetlabs -%}
+%%define upstream_name {{ metadata.puppetlabs_name }}
```

Comment:
yeah, I agree with that, we currently only explicitly cover openstack-puppet and puppetlabs.  This was the case previously with this script, so this is not a regression.  However, I could just rename this variable for now, and we can work on making it more robust in the next rev perhaps?